src/python/test/test_nextflow.py

```diff
@@ -188,7 +188,41 @@ def test_dx_build_nextflow_from_repository_default_metadata(self):
         details = applet.get_details()
         self.assertEqual(details["repository"], hello_repo_url)
 
+    @unittest.skipUnless(testutil.TEST_RUN_JOBS,
+                         'skipping tests that would run jobs')
+    def test_dx_build_nextflow_from_repository_with_extra_args(self):
+        pipeline_name = "hello"
+        hello_repo_url = "https://github.com/nextflow-io/hello"
+        extra_args = '{"name": "new_name", "title": "new title"}'
+        applet_id = run("dx build --nextflow --repository '{}' --extra-args '{}' --brief".format(hello_repo_url, extra_args)).strip()
+        applet = dxpy.DXApplet(applet_id)
+        desc = applet.describe()
+        self.assertEqual(desc["name"], "new name")
+        self.assertEqual(desc["title"], "new title")
+        self.assertEqual(desc["summary"], pipeline_name)
+
+class TestRunNextflowApplet(DXTestCaseBuildNextflowApps):
+
+    def test_dx_run_nextflow_with_additional_parameters(self):
+        pipeline_name = "hello"
+        applet_dir = self.write_nextflow_applet_directory(pipeline_name, existing_nf_file_path=self.base_nextflow_nf)
+        applet_id = json.loads(run("dx build --nextflow --json " + applet_dir))["id"]
+        applet = dxpy.DXApplet(applet_id)
+
+        job = applet.run({
+                         "nextflow_pipeline_params": "--input 'Printed_test_message'",
+                         "nextflow_top_level_opts": "-quiet"
```

Comment:
I will add a test for `nextflow_run_opts` separately as it's more difficult to test.